src/fastfilter.h

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2017 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_FAST_FILTER_H
+#define BITCOIN_FAST_FILTER_H
+
+#include "random.h"
+#include "serialize.h"
+#include <vector>
+
+class COutPoint;
+class uint256;
+
+#define REL_PRIME 27061
+/**
+ * FastFilter is a probabilistic filter.  The filter can answer whether an element
+ * definitely is NOT in the set, but only that an element is LIKELY in the set.
+ * This is similar to a Bloom filter, but much faster.
+ *
+ * This filter expects that the input elements have a random distribution (i.e. hashes)
+ *
+ * This class is thread-safe in the sense that simultaneous calls to insert and contains will not crash,
+ * but "inserts" may be lost.  However, if you are using this class as an in-ram filter before doing a more expensive
+ * operation, a lost insert may be acceptable.
+ */
```

Comment:
Maybe would be good to have a little written about why/how this filter is different from the usual bloom filter and what it's intended uses could be?

Reply:
how about: 
/**
 * FastFilter is a probabilistic filter.  The filter can answer whether an element
 * definitely is NOT in the set, but only that an element is LIKELY in the set.
 * This is similar to a Bloom filter, but much faster.
 *
 * This filter expects that the input elements have a random distribution (i.e. hashes), and so does not hash the
 * input again.  This is how it gains the majority of its performance improvement.
 *
 * This class can be used anywhere a Bloom filter is used so long as the input data is random.
 *
 * This class is thread-safe in the sense that simultaneous calls to insert and contains will not crash,
 * but "inserts" may be lost.  However, if you are using this class as an in-ram filter before doing a more expensive
 * operation, a lost insert may be acceptable.
 */

Reply:
that's good